src/switchroot/ostree-prepare-root.c

```diff
@@ -44,6 +44,25 @@
 
 #include "ostree-mount-util.h"
 
+static void
+die (void)
+{
+  if (getpid() == 1)
+    {
+      /* We're init - Give the user a chance to recover */
+      fprintf (stderr, "init: Entering recovery shell\n");
```

Comment:
> I think we need to make this part reliable :smile: And stay locked by default

I see your point, but I think having to rebuild might be a little inconvenient if things have already gone wrong.  How about a kernel command line option to enable this?  Something like `ostree.shell-on-boot-failure=1`?  Perhaps something a little less verbose?